diesel/src/sqlite/connection/mod.rs

```diff
@@ -132,6 +132,16 @@ impl SqliteConnection {
         }
         query.map(|_| ())
     }
+
+    #[doc(hidden)]
+    pub fn execute_pragma<ST, U>(&self, source: &str) -> QueryResult<Vec<U>>
+        where U: Queryable<ST, Sqlite>,
+              <SqliteConnection as Connection>::Backend: HasSqlType<ST>
+    {
+        Statement::prepare(&self.raw_connection, source)
+            .map(StatementIterator::new)
+            .and_then(Iterator::collect)
+    }
```

Comment:
select(...) will add a select statement  in front of the summited query. so this seems not to be a option here. One possibility would be to introduce a new pragma function, which adds pragma instead of select to the beginning of the query.